packages/core/plugins/core-plugins-server-internal/src/discovery/plugins_discovery.ts

```diff
@@ -64,10 +66,43 @@ export function discover({
       return typeof pluginPathOrError === 'string'
         ? createPlugin$(pluginPathOrError, log, coreContext, instanceInfo, nodeInfo)
         : [pluginPathOrError];
-    }),
-    shareReplay()
+    })
   );
 
+  const pluginPkgDiscovery$ = from(coreContext.env.repoPackages ?? EMPTY).pipe(
+    filter(
+      getPluginPackagesFilter({
+        oss: coreContext.env.cliArgs.oss,
+        examples: coreContext.env.cliArgs.runExamples,
+        paths: config.additionalPluginPaths,
+        parentDirs: config.pluginSearchPaths,
+      })
+    ),
+    map((pkg) => {
+      log.debug(`Successfully discovered plugin package "${pkg.id}"`);
+      const manifest = pluginManifestFromPluginPackage(
+        coreContext.env.packageInfo.version,
+        pkg.manifest
+      );
+      const initializerContext = createPluginInitializerContext({
+        coreContext,
+        opaqueId: Symbol(pkg.id),
+        manifest,
+        instanceInfo,
+        nodeInfo,
+      });
+
+      return new PluginWrapper({
+        path: pkg.directory,
+        manifest,
+        opaqueId: initializerContext.opaqueId,
+        initializerContext,
+      });
+    })
+  );
+
+  const discoveryResults$ = merge(fsDiscovery$, pluginPkgDiscovery$).pipe(shareReplay());
```

Comment:
```ts
merge(fsDiscovery$, pluginPkgDiscovery$)
```

I feel like we should be checking for duplicated from the two sources somehow here, unless we're adamant there's no risk of plugins being scanned by both (fs+package), wdyt?

Reply:
The filesystem scanning only finds legacy plugins with `kibana.json` files, and the package scanning only finds packages with `kibana.jsonc` files. `fsDiscovery$` won't ever produce duplicate values, though I'm find adding a distinct operator here.

Reply:
My naive question was: what about a folder with both `kibana.json` and `kibana.jsonc` for the same plugin?